File: CPP/FilmBioskop.cpp
#include "Film.cpp"

class FilmBioskop : public Film{
private:
    string studio;
    double hargaTiket;
    string jadwalTayang;

public:
    // Konstruktor ini membuat objek FilmBioskop dengan nilai awal kosong.
    FilmBioskop(): 
        Film(), studio(""), hargaTiket(0), jadwalTayang(""){
    }

    // Konstruktor ini mengisi data film beserta informasi penayangannya di bioskop.
    FilmBioskop(int id, string judul, string genre, int durasi, string sutradara, int tahunRilis, string studio, double hargaTiket, string jadwalTayang):
        Film(id, judul, genre, durasi, sutradara, tahunRilis), studio(studio), hargaTiket(hargaTiket), jadwalTayang(jadwalTayang){
    }

    // Method ini mengembalikan nama studio pemutaran.
    string getStudio() const{
        return studio;
    }

    // Method ini mengembalikan harga tiket film.
    double getHargaTiket() const{
        return hargaTiket;
    }

    // Method ini mengembalikan jadwal tayang film.
    string getJadwalTayang() const{
        return jadwalTayang;
    }

    // Method ini mengubah nama studio pemutaran.
    void setStudio(string studio){
        this->studio = studio;
    }

    // Method ini mengubah harga tiket film.
    void setHargaTiket(double hargaTiket){
        this->hargaTiket = hargaTiket;
    }

    // Method ini mengubah jadwal tayang film.
    void setJadwalTayang(string jadwalTayang){
        this->jadwalTayang = jadwalTayang;
    }

    // Method ini menampilkan seluruh data film beserta informasi bioskopnya.
    void tampilkanData() const override{
        Film::tampilkanData();
        cout << "Studio : " << studio << endl;
        cout << "Harga Tiket : Rp " << hargaTiket << endl;
        cout << "Jadwal Tayang : " << jadwalTayang << endl;
    }
};

File: Java/FilmBioskop.java
public class FilmBioskop extends Film{
    private String studio;
    private double hargaTiket;
    private String jadwalTayang;

    // Konstruktor ini membuat objek FilmBioskop dengan nilai awal kosong.
    public FilmBioskop(){
        super();
        this.studio = "";
        this.hargaTiket = 0;
        this.jadwalTayang = "";
    }

    // Konstruktor ini mengisi data film beserta informasi penayangannya di bioskop.
    public FilmBioskop(int id, String judul, String genre, int durasi, String sutradara, int tahunRilis, String studio, double hargaTiket, String jadwalTayang){
        super(id, judul, genre, durasi, sutradara, tahunRilis);
        this.studio = studio;
        this.hargaTiket = hargaTiket;
        this.jadwalTayang = jadwalTayang;
    }

    // Method ini mengembalikan nama studio pemutaran.
    public String getStudio(){
        return studio;
    }

    // Method ini mengembalikan harga tiket film.
    public double getHargaTiket(){
        return hargaTiket;
    }

    // Method ini mengembalikan jadwal tayang film.
    public String getJadwalTayang(){
        return jadwalTayang;
    }

    // Method ini mengubah nama studio pemutaran.
    public void setStudio(String studio){
        this.studio = studio;
    }

    // Method ini mengubah harga tiket film.
    public void setHargaTiket(double hargaTiket){
        this.hargaTiket = hargaTiket;
    }

    // Method ini mengubah jadwal tayang film.
    public void setJadwalTayang(String jadwalTayang){
        this.jadwalTayang = jadwalTayang;
    }

    @Override
    // Method ini menampilkan seluruh data film beserta informasi bioskopnya.
    public void tampilkanData(){
        super.tampilkanData();
        System.out.println("Studio : " + studio);
        System.out.println("Harga Tiket : Rp " + hargaTiket);
        System.out.println("Jadwal Tayang : " + jadwalTayang);
    }
}
